packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -66,43 +166,28 @@ def vip_test(dcos_api_session, r):
         host2 = agents[1]
     log.debug('host1 is is: {}'.format(host1))
     log.debug('host2 is is: {}'.format(host2))
-    origin_app, app_uuid = get_test_app_in_docker()
-    origin_app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': r.vip}
-    origin_app['container']['docker']['network'] = r.vipnet
-    origin_app['mem'] = 0
-    origin_app['cpu'] = 0
-    if r.vipnet == 'USER':
-        origin_app['ipAddress'] = {'networkName': 'dcos'}
-    if r.vipnet == 'HOST':
-        origin_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
-                            '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py $PORT0'
-        origin_app['container']['docker']['portMappings'][0]['hostPort'] = 0
-        origin_app['container']['docker']['portMappings'][0]['containerPort'] = 0
-        origin_app['container']['docker']['portMappings'][0]['labels'] = {}
-        origin_app['portDefinitions'] = [{'labels': {'VIP_0': r.vip}}]
-
-    origin_app['constraints'] = [['hostname', 'CLUSTER', host1]]
-
-    proxy_app = get_test_app_in_docker()[0]
-    proxy_app['container']['docker']['network'] = r.proxynet
-    proxy_app['constraints'] = [['hostname', 'CLUSTER', host2]]
-    proxy_app['mem'] = 0
-    proxy_app['cpu'] = 0
-    if r.proxynet == 'USER':
-        proxy_app['ipAddress'] = {'networkName': 'dcos'}
-    if r.proxynet == 'HOST':
-        proxy_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
-                           '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py $PORT0'
-        proxy_app['container']['docker']['portMappings'][0]['hostPort'] = 0
-        proxy_app['container']['docker']['portMappings'][0]['containerPort'] = 0
 
-    returned_uuid = None
+    origin_app, app_uuid = vip_app(r.num, r.container, r.vipnet, host1, r.vip)
+    # allow to run on both public and private agents
+    origin_app['acceptedResourceRoles'] = ["*", "slave_public"]
+    proxy_app, _ = vip_app(r.num, r.container, r.proxynet, host2, None)
+    # allow to run on both public and private agents
+    proxy_app['acceptedResourceRoles'] = ["*", "slave_public"]
 
+    returned_uuid = None
     with contextlib.ExitStack() as stack:
+        # try to avoid thundering herd
+        time.sleep(random.randint(0, 30))
```

Comment:
Is this wise? Why are we handicapping our tests? Can the marathon/mesos really not handle so many app POST's?